src/OpenTelemetry.Instrumentation.AspNet.TelemetryHttpModule/ActivityHelper.cs

```diff
@@ -64,22 +65,38 @@ public static bool HasStarted(HttpContext context, out Activity? aspNetActivity)
         PropagationContext propagationContext = textMapPropagator.Extract(default, context.Request, HttpRequestHeaderValuesGetter);
 
         KeyValuePair<string, object?>[]? tags;
```

Comment:
Why not replace `KeyValuePair<string, object?>[]?` with `List<KeyValuePair<string, object?>>?` for dynamic tag management, which eliminates the need for manual index handling in arrays, making the code more flexible and cleaner?

Reply:
@ysolomchenko, I would say that it is due to performance. If you think that it is worth to change without loosing it, please create PR. Some benchmarks will be needed.

Reply:
@Kielek, Yes, the speed will indeed drop by about five times (39 ns vs. 7 ns). However, since we're talking about nanoseconds, I'm not sure how critical that is. It might not be noticeable in real-world scenarios. If this difference is truly important, then we should consider abandoning this change.

Reply:
It is on the hot-path for each asp.net request. If it will be 5-6 times slower, we should avoid switching to the list version.